mypyc/genops.py

```diff
@@ -1998,9 +1998,26 @@ def make_for_loop_generator(self,
         # Default to a generic for loop.
         expr_reg = self.accept(expr)
         for_obj = ForIterable(self, index, body_block, loop_exit, line, nested)
-        for_obj.init(expr_reg)
+        item_type = self._analyze_iterable_item_type(expr)
+        item_rtype = self.type_to_rtype(item_type)
+        for_obj.init(expr_reg, item_rtype)
         return for_obj
 
+    def _analyze_iterable_item_type(self, expr: Expression) -> Type:
```

Comment:
Add a docstring to this one, please. (Including a mention of where in mypy it is basically copied/modified from)